### PR TITLE
Update cheat codes for Melee

### PR DESCRIPTION
**Pull request by Stevoisiak**
Also made minor formatting changes

### CODE REVIEW

File: Data/Sys/GameSettings/GALE01r0.ini [hunk 1]
```diff
@@ -182,20 +227,248 @@ $AI clone, Don't use in story modes or with zelda/sheik. (CPU level
...
+$Player 3 gets two AI clones
 04002F08 00000002
-$player 4 gets two AI clones.
+
+$Player 4 gets two AI clones
 04002F0C 00000002
 
+[Gecko]
+# General Codes
+Unlock All Characters and Stages [Datel]
```
Comment (JosJuice) [inline Data/Sys/GameSettings/GALE01r0.ini]: Why is there no `$` here?

Comment (Stevoisiak) [inline Data/Sys/GameSettings/GALE01r0.ini]: Because I'm silly and missed that somehow. :P

**Comment by jefflongo** [comment]
While you're doing this, would you mind replacing the True Widescreen gecko code with this more accurate one?  It's also made by Internet Explorers and it's for 1.02.

```
C2021ABC 00000002
39C00001 38600006
60000000 00000000
C236A4A8 00000007
C03F0034 2C0E0001
41820024 3C004260
90010030 3C00421C
90010034 C0010030
EC210032 C0010034
EC210024 39C00000
281E0000 00000000
```

**Comment by Stevoisiak** [comment]
@jefflongo What's different about that code?

**Comment by jefflongo** [comment]
https://www.reddit.com/r/SSBM/comments/31lc8t/my_melee_looks_funny_an_in_depth_breakdown_of/

**Comment by jefflongo** [comment]
Also these 3 gecko codes would be nice additions as well.  All of these are Netplay safe which makes them really awesome.

NTSC stock icons can be big and in the way.  PAL's are smaller yet the same thing. 
```
$PAL Stock Icons [Achilles]
C22F9A28 00000004
3C003F59 6000999A
901D002C 901D0030
3C00C1B0 60000000
60000000 00000000
```

This one is nice so you can see how often you got L cancel's in a match
```
$L Cancel % in Results Screen [Wooggle]
C208D600 00000009
8A830678 3DC08000
61CE45D4 1EB40008
7DCEAA14 81EE0000
39EF0001 91EE0000
3A0EFFFC 82300000
7E717BD6 3E408045
62523DA4 1EB40E90
7E52AA14 92720000
3880FFFF 00000000
C208D6A4 00000009
8A830678 3DC08000
61CE45D4 1EB40008
7DCEAA14 81EE0000
3A0EFFFC 82300000
3A310064 92300000
7E717BD6 3E408045
62523DA4 1EB40E90
7E52AA14 92720000
EC010024 00000000
04040BBC 60000000
20C61DB8 16200D20
06C61DB8 00000014
1620151A 200C2024
20312026 2028202F
1A210300 00000000
020045D0 000F0000
E2000001 00000000
```

I don't know if you already have one like this but this one swaps the value for rumble on or off (default it's on and some people find that annoying since most players play with it off).
```
$Disable Rumble
043D4A70 00000000
```

**Comment by Stevoisiak** [comment]
Are any of those codes specific to a certain version of the game?

**Comment by jefflongo** [comment]
These all are for NTSC 1.02.  Not sure if they work on any other version of the game.

**Comment by Stevoisiak** [comment]
@jefflongo Added the codes.

**Comment by rohlem** [comment]
Idk if y'all would prefer an issue about this, but I figured if I'm quick enough...
Most of the AR-codes for NTSC Revision 0 (in GALE01r0.ini) make Dolphin crash. Enabling Config->Interface->Use Panic Handlers makes an info-box appear that shows "Unknown instruction 00000000 at PC = 80085320 last_PC = 81200d08 LR = 80031c70", which is obviously the cause.
This was actually noticed a long time ago on the forums: https://forums.dolphin-emu.org/Thread-gc-super-smash-bros-melee?pid=156373#pid156373
Removing all lines with the second column 08000000 fixes all of them.

**Comment by Stevoisiak** [comment]
@rohlem I'm assuming those codes work fine on console?

**Comment by rohlem** [comment]
@Stevoisiak I'll have to verify that right away, however they work as intended when the "corrupt" line is emitted.
Also, there is no equivalent line in any of the other version's AR-codes. I have only minimal knowledge about AR, but it seems odd that only revision 0 would require this extra step. (Maybe they were copied accidentally, being a part of some bigger loader-code?)

**Comment by Shadoxfix** [comment]
LGTM once the melee 1.00 codes are confirmed working.

**Comment by rohlem** [comment]
@Shadoxfix **I've just tested all of the v1.0 codes on Dolphin, and can confirm none of them crash** after the intro as they used to. *(All crashes/freezes that still occur are due to invalid reads/writes when accessing non-playable character data and such.)*

@Stevoisiak Two of the codes still seem faulty:
* "Maximize Statistics" has the side effect of **overwriting all custom names with some weird symbols that cause a soft-lock** (music keeps playing, game stops responding to inputs) when opening the name selection. As there is no way to delete names only, **this basically bricks your save file**. *Personally, I'd suggest to just remove that code altogether, since the data it modifies isn't really checked by the game in any way, which makes it rather pointless to begin with.*
* "AI clone" straight out **doesn't do anything**. I have however found a **different version** [here](http://arcentral.net/ssbm/ssbmV10/v10partner.html) (named "Each player gets a CPU clone that fights for them"), **which works flawlessly**. The only difference I can spot is that 3 addresses (I assume) are altered. *My guess is the current .ini - version is intended for a different version of the game. I've tried v1.2 and PAL though, without success (or any visible result whatsoever).*

**Comment by Stevoisiak** [comment]
@rohlem I've updated the PR to remove the Maximize Statistics code and update the AI Clone code.

**Comment by rohlem** [comment]
@Stevoisiak Confirming the AI Clone code works correctly now.
Would you mind putting the phrase "AI Clone" back into its name though? Otherwise users might not draw the connection to the following "Player X gets Y AI clones" codes it's meant to be used with.

**Comment by Stevoisiak** [comment]
@rohlem I'd prefer to keep it as CPU so users know it's a different code. [It's also what the code's creator called it](http://arcentral.net/ssbm/ssbmV10/v10partner.html). However, I can remove "AI" from the other codes.

**Comment by rohlem** [comment]
@Stevoisiak fair enough, no more objections. *Thank you for hearing me out this quickly!*

**Pull request merged**